pkg/backporter/backporter.go

```diff
@@ -553,30 +558,118 @@ func CreateCloneHandler(client bugzilla.Client, allTargetVersions sets.String, m
 			handleError(w, err, fmt.Sprintf("unable to fetch bug details- Bug#%d", bugID), http.StatusNotFound, endpoint, bugID, m)
 			return
 		}
-		// Create a clone of the bug
-		cloneID, err := client.CloneBug(bug)
+		if !allTargetVersions.Has(req.FormValue("release")) {
+			handleError(w, fmt.Errorf("invalid argument - %s is not a valid TargetRelease, must be one of %v", req.FormValue("release"), allTargetVersions.List()), fmt.Sprintf("invalid argument - %s is not a valid TargetRelease, must be one of %v", req.FormValue("release"), allTargetVersions.List()), http.StatusBadRequest, endpoint, bugID, m)
+			return
+		}
+
+		// Get clones and sort them
+		clones, err := client.GetAllClones(bug)
 		if err != nil {
-			handleError(w, err, "clone creation failed", http.StatusInternalServerError, endpoint, bugID, m)
+			handleError(w, err, fmt.Sprintf("unable to convert \"ID\" parameter from string to int: %s", req.FormValue("ID")), http.StatusBadRequest, req.URL.Path, 0, m)
 			return
 		}
-		targetRelease := bugzilla.BugUpdate{
-			TargetRelease: []string{
-				req.FormValue("release"),
-			},
+		sort.SliceStable(clones, func(i, j int) bool {
+			if len(clones[i].TargetRelease) == 0 && len(clones[j].TargetRelease) == 0 {
+				return false
+			} else if len(clones[i].TargetRelease) == 0 {
+				return true
+			} else if len(clones[j].TargetRelease) == 0 {
+				return false
+			}
+			return clones[i].TargetRelease[0] < clones[j].TargetRelease[0]
+		})
+		toCloneMajorMinorRelease, err := getMajorMinorRelease(req.FormValue("release"))
+		if err != nil {
+			handleError(w, err, fmt.Sprintf("%s cannot be parsed as release, must be of the form X.X.X", req.FormValue("release")), http.StatusBadRequest, endpoint, bugID, m)
+			return
+		}
+		var sourceBug *bugzilla.Bug
+		var sourceBugMajorMinorRel string
+		for _, clone := range clones {
+			if len(clone.TargetRelease) == 0 || clone.TargetRelease[0] == "---" {
+				continue
+			}
+			cloneMajorMinorRel, err := getMajorMinorRelease(clone.TargetRelease[0])
+			if err != nil {
+				handleError(w, err, fmt.Sprintf("%s cannot be parsed as release, must be of the form X.X.X", clone.TargetRelease[0]), http.StatusBadRequest, endpoint, bugID, m)
+				return
+			}
+			if toCloneMajorMinorRelease == cloneMajorMinorRel {
+				handleError(w, err, fmt.Sprintf("clone for major release %s already exists", clone.TargetRelease[0]), http.StatusBadRequest, endpoint, bugID, m)
```

Comment:
Why can the user ask to clone to a target release for which a bug already exists?

Reply:
The user would ideally not be allowed to do it from the UI, since it would not be available in the dropdown.
This error would only be thrown if the user somehow calls the endpoint directly